frappe/email/doctype/email_account/email_account.py

```diff
@@ -410,6 +419,9 @@ def set_thread(self, communication, email):
 
 		parent = self.find_parent_from_in_reply_to(communication, email)
 
+		if not frappe.safe_eval(self.condition, None, get_context(communication)):
```

Comment:
put this in the try block so that even if condition fails, it will pull the email or log an error in Error Log